src/devices/Nrf24l01/Nrf24l01.cs

```diff
@@ -0,0 +1,773 @@
+        /// </summary>
+        /// <returns>Address</returns>
+        internal Span<byte> ReadTxAddress()
+        {
+            return WriteRead(Command.NRF_R_REGISTER, Register.NRF_TX_ADDR, 5);
+        }
+
+        /// <summary>
+        /// Set Working Channel
+        /// </summary>
+        /// <param name="channel">From 0 to 127</param>
+        internal void SetChannel(byte channel)
+        {
+            if (channel < 0 || channel > 127)
+            {
+                throw new ArgumentOutOfRangeException("Channel needs to be in the range of 0 to 127!");
+            }
+
+            _gpio.Write(_ce, PinValue.Low);
+
+            Write(Command.NRF_W_REGISTER, Register.NRF_RF_CH, channel);
+
+            _gpio.Write(_ce, PinValue.High);
+        }
+
+        /// <summary>
+        /// Read Working Channel
+        /// </summary>
+        /// <returns>Channel</returns>
+        internal byte ReadChannel()
+        {
+            Span<byte> readData = WriteRead(Command.NRF_R_REGISTER, Register.NRF_RF_CH, 1);
+
+            return readData[0];
+        }
+        #endregion
+
+        #region sensor operation
+        internal void Write(Span<byte> writeData)
+        {
+            Span<byte> readBuf = stackalloc byte[writeData.Length];
+
+            _sensor.TransferFullDuplex(writeData, readBuf);
+        }
+
+        internal void Write(Command command, Register register, byte writeByte)
+        {
+            Span<byte> writeBuf = stackalloc byte[2] { (byte)((byte)command + (byte)register), writeByte };
+            Span<byte> readBuf = stackalloc byte[2];
+
+            _sensor.TransferFullDuplex(writeBuf, readBuf);
+        }
+
+        internal void Write(Command command, Register register, Span<byte> writeData)
+        {
+            Span<byte> writeBuf = stackalloc byte[1 + writeData.Length];
+            Span<byte> readBuf = stackalloc byte[1 + writeData.Length];
+
+            writeBuf[0] = (byte)((byte)command + (byte)register);
+            for (int i = 0; i < writeData.Length; i++)
+            {
+                writeBuf[1 + i] = writeData[i];
+            }
+
+            _sensor.TransferFullDuplex(writeBuf, readBuf);
+        }
+
+        internal Span<byte> WriteRead(Command command, Register register, int dataLength)
+        {
+            Span<byte> writeBuf = stackalloc byte[1 + dataLength];
+            Span<byte> readBuf = new byte[1 + dataLength];
+
+            writeBuf[0] = (byte)((byte)command + (byte)register);
+
+            _sensor.TransferFullDuplex(writeBuf, readBuf);
+
+            return readBuf.Slice(1);
+        }
+
+        internal Span<byte> WriteRead(Command command, Register register, Span<byte> writeData)
+        {
+            Span<byte> writeBuf = stackalloc byte[1 + writeData.Length];
+            Span<byte> readBuf = new byte[1 + writeData.Length];
+
+            writeBuf[0] = (byte)((byte)command + (byte)register);
+            for (int i = 0; i < writeData.Length; i++)
```

Comment:
This should be `writeData.CopyTo(writeBuf.Slice(1));`  You'll get the best performance that way.